pkg/controllers/nodeclass/status/images.go

```diff
@@ -147,9 +171,54 @@ func imageVersionsUnready(nodeClass *v1alpha2.AKSNodeClass) bool {
 }
 
 // Handles case 4: check if the maintenance window is open
-func isMaintenanceWindowOpen() bool {
-	// TODO: need to add the actual logic for handling maintenance windows once it is in the ConfigMap.
-	return true
+// TODO (charliedmcb): remove nolint on gocyclo. Added for now in order to pass "make verify"
+// I think the best way to get rid of gocyclo is to break the section retrieving the maintenance window
+// range from the ConfigMap into its own helper function using channel as a parameter.
+// nolint: gocyclo
+func (r *NodeImageReconciler) isMaintenanceWindowOpen(ctx context.Context) (bool, error) {
```

Comment:
It feels like some form of caching would be appropriate here, but it is probably fine to revisit this later.
(I am assuming getting ConfigMap on every call is Ok for now - as long as we use caching client.)

Reply:
So, this is actually an instance where I think we need to be rechecking this about every 5 minutes, to catch updated to it. Another pattern we were considering was to look at setting up a watch, but we want this catch updates pretty quickly.

Reply:
We can (and possibly _should_?) set up a watch to just tell us when this configmap is updated and trigger a reconcile?

You can see how ASO does this here: https://github.com/Azure/azure-service-operator/blob/main/v2/internal/controllers/controller_resources.go#L343

Reply:
Ok I see we have requeue every 5m so it should be fine, though it might one day be better to just have this be a watch on the configmap I'm not super hung up on it (it gets you caching too which I guess is nice)

Reply:
also this is tricky w/ the fact we have 2 API servers in some cases so the controller-runtime clients are all pointed at the overlay one, so a watch via controller-runtime is actually harder than it sounds here.